csrc/ir/internal_nodes.h

```diff
@@ -2327,4 +2327,77 @@ class LinearOp : public Expr {
   }
 };
 
+// SDPA node with same functionality as F.scaled_dot_product_attention
+// https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html
+class SdpaOp : public Expr {
+ public:
+  using Expr::Expr;
+
+  SdpaOp(
+      IrBuilderPasskey,
+      Val* out,
+      Val* query,
+      Val* key,
+      Val* value,
+      Val* attn_mask,
+      double dropout_p,
+      bool is_causal,
+      std::optional<double> scale);
```

Comment:
Same. I think this should be a scalar `Val*`, since it will commonly be a function of the input dimensions.